test/functional/wiki_controller_test.rb

```diff
@@ -504,6 +504,18 @@ def teardown
     assert_response :success
   end
 
+  test "redirect to non-existent page fails gracefully; no redirect" do
+    wiki = node(:wiki_page)
+    slug = wiki.path.gsub('/wiki/', '')
+    user = rusers(:bob)
+    tagname = "redirect:nonsense"
+    wiki.add_tag(tagname, user)
```

Comment:
Here we should not expect this to succeed -- see this line, it will return `[false, tag]` and we should be able to assert:

```ruby
assert_false wiki.add_tag(tagname, user)[0]
```

Make sense? We want to test that it rejects this tagname.